pkg/controller/beat/common/config.go

```diff
@@ -13,66 +13,92 @@ import (
+	username, password, err := association.ElasticsearchAuthSettings(client, &associated)
 	if err != nil {
-		return err
+		return settings.NewCanonicalConfig(), err
 	}
 
-	esOutput := settings.MustCanonicalConfig(
-		map[string]interface{}{
-			"output.elasticsearch": map[string]interface{}{
-				"hosts":    []string{associated.AssociationConf().GetURL()},
-				"username": username,
-				"password": password,
-			},
-		})
-
-	if err := cfg.MergeWith(esOutput); err != nil {
-		return err
+	esOutput := map[string]interface{}{
+		"output.elasticsearch": map[string]interface{}{
+			"hosts":    []string{associated.AssociationConf().GetURL()},
+			"username": username,
+			"password": password,
+		},
 	}
 
 	if associated.AssociationConf().GetCACertProvided() {
-		if err := cfg.MergeWith(settings.MustCanonicalConfig(
-			map[string]interface{}{
-				"output.elasticsearch.ssl.certificate_authorities": path.Join(CAMountPath, CAFileName),
-			})); err != nil {
-			return err
-		}
+		esOutput["output.elasticsearch.ssl.certificate_authorities"] = []string{path.Join(certificatesDir(&associated), CAFileName)}
 	}
 
-	return nil
+	return settings.NewCanonicalConfigFrom(esOutput)
+}
+
+// BuildKibanaConfig builds on optional Kibana configuration for dashboard setup and visualizations.
+func BuildKibanaConfig(client k8s.Client, associated beatv1beta1.BeatKibanaAssociation) (*settings.CanonicalConfig, error) {
+	if !associated.AssociationConf().IsConfigured() {
+		return settings.NewCanonicalConfig(), nil
+	}
+
+	username, password, err := association.ElasticsearchAuthSettings(client, &associated)
+	if err != nil {
+		return settings.NewCanonicalConfig(), err
+	}
+
+	kibanaCfg := map[string]interface{}{
+		"setup.dashboards.enabled": true,
+		"setup.kibana": map[string]interface{}{
+			"host":     associated.AssociationConf().GetURL(),
+			"username": username,
+			"password": password,
+		},
+	}
+
+	if associated.AssociationConf().GetCACertProvided() {
+		kibanaCfg["setup.kibana.ssl.certificate_authorities"] = []string{path.Join(certificatesDir(&associated), CAFileName)}
+	}
+	return settings.NewCanonicalConfigFrom(kibanaCfg)
 }
 
 func buildBeatConfig(
 	log logr.Logger,
 	client k8s.Client,
 	beat beatv1beta1.Beat,
-	defaultConfig *settings.CanonicalConfig,
+	defaultConfig DefaultConfig,
 ) ([]byte, error) {
 	cfg := settings.NewCanonicalConfig()
 
-	if err := setOutput(cfg, client, &beat); err != nil {
+	outputCfg, err := buildOutputConfig(client, beatv1beta1.BeatESAssociation{Beat: &beat})
+	if err != nil {
+		return nil, err
+	}
+	err = cfg.MergeWith(outputCfg, defaultConfig.Managed)
+	if err != nil {
 		return nil, err
 	}
-
 	// use only the default config or only the provided config - no overriding, no merging
 	userConfig := beat.Spec.Config
 	if userConfig == nil {
-		if err := cfg.MergeWith(defaultConfig); err != nil {
+		if err := cfg.MergeWith(defaultConfig.Unmanaged); err != nil {
```

Comment:
nit: it's not in your change but I just realized that in the debug log below we probably want to include the beat name/ns